aten/src/ATen/native/cuda/block_reduce.cuh

```diff
@@ -10,6 +10,7 @@ namespace native {
 namespace cuda_utils {
 
 constexpr int kCUDABlockReduceNumThreads = 512;
+constexpr int kCUDABlockReduceMaxThreads = C10_WARP_SIZE * C10_WARP_SIZE;
```

Comment:
Can you explain your logic? The function does two reductions, each of which divides the width by `C10_WARP_SIZE`. So the max size should be `C10_WARP_SIZE` squared.

Reply:
Oh, you mean that the device `maxThreadsPerBlock` is likely to be 1024? I meant this to reflect the limitation of the algorithm, not hardware. If you call with more threads than this value, the algorithm silently fails.

Reply:
Yeah, in terms of algorithm limitations you are right, both on ROCm and on CUDA number of threads per block cannot be larger anyway (on ROCm it can only be smaller, as this evaluates to 4096, and maxThreadsPerBlock is 1024( , but it's ok to leave it as is. Can you please leave a comment that it's an algorithm limitation?